adhocteam/Head-Start-TTADP
# Cm 21 add notifications

**Description of change**
Adds four notifications:
1. When you are added as a collaborator to an Activity Report
2. When an Activity report is submitted for approval
3. When a manager requests changes
4. When an activity report is approved

These notifications are currently behind a feature flag and will not be sent unless the SENDNOTIFICATIONS environmental variable is set. Until we have an smtp gateway to send email from, these features will remain disabled in production.

**How to test**
1. Add the following variables to your .env file
```
FROM_EMAIL_ADDRESS=ttasmarthub@test.gov
SEND_NOTIFICATIONS=true
SMTP_HOST=192.168.x.xxx (should be your actual IP address for your computer)
SMTP_USER=''
SMTP_PASS=''
SMTP_PORT=1025
SMTP_SECURE=false
```
2. run `docker run -d -p 1080:1080 -p 1025:1025 --name mailcatcher schickling/mailcatcher`. This will start a local smtp server that will display all the mail sent out.
3. set line 24 in `src/lib/mailer/index.js` to true for testing It should be `const send = true;` for local testing.
3. Open your web browser and go to localhost:1080 to see the mailcatcher UI
4. Open the app in your browser and start a new activity report.
5. Add a collaborator and click "save draft"
6. You should get an email in mailcatcher notifying the new collaborator that they were added as a collaborator (you may need to refresh). If you add multiple collaborators, you should see an email go to each of them.
7. complete the form and click submit
8. You should see a mail to the approving manager notifying them that they have a report to approve
9. as the approving manager, open the report and request changes.
10. You should see an email to the author and all collaborators saying there were changes requested along with the managers comments.
11. As the manager approve the report
12. you should see an email to the author and collaborators showing their report was approved.

**Issue(s)**
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-19
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-20
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-21

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
